src/main/java/com/booleanuk/core/Exercise.java

```diff
@@ -11,35 +11,67 @@ public class Exercise {
 
     public void stepOne() {
         // TODO: 1. Write a for loop that adds the numbers 0 to 3 to the numsZeroToThree array
+        int[] arr = new int[4];
 
+        for (int i = 0; i <= 3 ; i++) {
+            arr[i] = i;
+
+        }
+        this.numsZeroToThree = arr;
 
     }
 
     public void stepTwo() {
         // TODO: 2. Write a for loop that adds the numbers 5 to 10 to the numsFiveToTen array
+        int[] arr = new int[6];
+
+        for (int i = 5; i <= 10 ; i++) {
+            arr[i-5] = i;
 
+        }
+        this.numsFiveToTen = arr;
 
     }
 
     public void stepThree() {
         // TODO: 3. Write a for loop that adds the numbers 3 to 0 (in that order) to the countdown array
+        int[] arr = new int[4];
+        int num=3;
+        for (int i = 0; i <= 3 ; i++) {
+            arr[i] = num;
+            num--;
 
+        }
+        this.countdown = arr;
 
     }
 
     public boolean stepFour(int num) {
         // TODO: 6. Write a for loop that checks if num is in the favouriteNumbers array
+        int numberIn=0;
+        boolean favNumberInArray =false;
+        for (int i = 0; i <= favouriteNumbers.length-1 ; i++) {
+            numberIn=favouriteNumbers[i];
+            if (numberIn==num)favNumberInArray=true;
+
+        }
```

Comment:
Could be simplifed:

```
public boolean stepFour(int num) {
    for (int i = 0; i < favouriteNumbers.length; i++) {
        if (favouriteNumbers[i] == num) {
            return true;
        }
    }
    return false;
}
```

Reply:
You can apply this approach to your other answers to improve readability and efficency